src/main/resources/transformation/metafix-wip.fix

```diff
@@ -0,0 +1,27 @@
+/*
+  <entity name="TitleOther">
+    <entity name="title[]" flushWith="record">
+      <data name="" source="246?[ 345678].a">
+        <replace pattern="\.$" with="" />
+      </data>
+    </entity>
+  </entity>
+	*/
+
+# TODO: Das schreiben in einen tieferliegendes Array funktioniert anscheinend nicht.
```

Comment:
I'm working on exactly the same problem right now. My (preliminary) solution:

```
do list(path: "246?[ 345678]", "var": "$i")
  if exists("@title_other")
    add_field("@title_other_array", "true")
  else
    set_array("@title_other")
  end
  replace_all("$i.a", "\\.$", "")
  copy_field("$i.a", "@title_other")
end
if exists("@title_other_array")
  copy_field("@title_other", "TitleOther.title[]")
else
  copy_field("@title_other", "TitleOther.title")
end
```

Does this pattern work here as well?

Reply:
Clarification: The problem is not that it's a nested array, but that it may be an array or a string.

Reply:
I am not sure if I understand the problem correctly. What can be an array or an string?
If I copy the field on the "root" level as `"TO []"` The result seems right and seems always to be an array.
If I copy the field to a hierarchy-level below as `"TitleOther.title[]"` it is inconsistent if it is an array or an string-element.
Similar if I copy the results of the list-bind straight to `"TitleOther.title[]"` the result has index-numbers if more than one values come in, see:
https://github.com/hbz/limetrans/pull/79/commits/89f6bbe3a63c8c2c98170209a3b6fb9a58d24919

Reply:
Yes, it's actually a combination of both: If we want to create a field that can be an array or a string (depending on the number of occurrences of the source field), then we can't simply map the source field into the nested target field. We have to use a temporary top-level field (`@title_other`) _and_ keep track of whether the source field occurred multiple times (`@title_other_array`).

Can you try this approach as outlined above?